### PR TITLE
Add conflict status to OpenAPI

### LINKED ISSUE #50: Update OpenAPI for Conflict HTTP Status
As a developer creating clients against the OSCAL REST API, I want to receive a more specific conflict HTTP status code for operations that have the potential to have a conflict, particularly around mismatch of UUIDs, so that my application can more easily react to those responses.

### Acceptance Criteria
- The OSCAL REST API includes HTTP conflict status for relevant operations

### PR DESCRIPTION
**Pull request by kperk103**
This adds conflict HTTP status under `PATCH` and `PUT` operations.
I was unsure how to know the cause of the conflict so the message
for each conflict status says "Conflict with current state." If
there is anything I can look at to know how to make the message
better/more specific please let me know.

Closes #50

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -211,6 +211,9 @@ paths:
         405:
           description: Validation exception
           content: {}
+        409:
+          description: Conflict with current state
+          content: {}
```
Comment (tuckerzp) [inline openapi.yaml]: I feel like using an empty content tag here is not necessary, From what I can tell, it adds a little `media type` tag that does not provide any functionality. I don't think you should remove it everywhere we do this, but I am somewhat against adding that if not necessary.

**Review by tuckerzp — CHANGES_REQUESTED** [review]
I like the use of a `409` error here. I have not looked into what would fit best a ton, but that seems to fit very well. As far as the `description` tag goes, it seems like we never get _too_ specific with the errors. I am leaning towards `Conflict with current state` being good enough,

File: openapi.yaml (hunk 1 above)
Comment (kperk103) [reply openapi.yaml]: I agree. I removed the empty content tag from the 409 status codes that I had previously included.

File: openapi.yaml [hunk 2]
```diff
@@ -248,6 +250,8 @@ paths:
           description: Bad Request
         415:
           description: Unsupported media type
+        409:
+          description: Conflict with current state
```
Comment (Bronstrom) [inline openapi.yaml]: A little nitpicky, but I think we should order these error responses in numerical order. It seems like 404 appears before 400 a few times too — while unrelated to 409, I would suggest we swap these in this PR as well.

**Review by Bronstrom — CHANGES_REQUESTED** [review]
This is looking great so far! I have one suggestion:

**Comment by Bronstrom** [comment]
Also, somewhat out of scope of this issue and a small detail, but it seems like our choice to include `content: {}` is ambiguous throughout the file, while some 4XX error responses to `PUT` and `PATCH` operations include this while others don't. Like @tuckerzp in his request, I don't believe these provide any value at all so these could just be removed. Although being unrelated to the 409 error, perhaps this is something that should be addressed in a separate PR?

**Comment by kperk103** [comment]
> Also, somewhat out of scope of this issue and a small detail, but it seems like our choice to include `content: {}` is ambiguous throughout the file, while some 4XX error responses to `PUT` and `PATCH` operations include this while others don't. Like @tuckerzp in his request, I don't believe these provide any value at all so these could just be removed. Although being unrelated to the 409 error, perhaps this is something that should be addressed in a separate PR?

I definitely agree. I can work on that and create a PR for it. The empty content tags are also sometimes in operations other than `PUT` and `PATCH`. Do you recommend I remove those as well for consistency?

**Comment by Bronstrom** [comment]
> I definitely agree. I can work on that and create a PR for it. The empty content tags are also sometimes in operations other than PUT and PATCH. Do you recommend I remove those as well for consistency?

Yeah, that makes more sense. I would just remove them everywhere they appear in the file. Thanks!

**Review by Bronstrom — APPROVED** [review]

**Review by tuckerzp — APPROVED** [review]
Thanks for your work on this!

**Pull request merged**